### PR TITLE
[Twilio Adapter] Replace interfaces with classes

### PR DESCRIPTION
**Pull request by ceciliaavila**
## Proposed Changes
Replace interfaces with classes to allow versioning.

## Detailed changes
- Remove _ITwilioClient_ interface.
- Convert interface _ITwilioAdapterOptions_ in a class 
- Rename TwilioApi class as TwilioClientWrapper and made its methods virtual.
- Adjust TwilioAdapter class and unit tests to these changes.
- Replace explicit types with `var`

### CODE REVIEW

File: tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioHelperTests.cs [hunk 1]
```diff
@@ -106,8 +106,8 @@ public void QueryStringToDictionary_Should_Return_Dictionary_With_Valid_Query()
             var builder = new StringBuilder(ValidationUrlString);
 
             var bodyString = File.ReadAllText(Directory.GetCurrentDirectory() + @"\files\NoMediaPayload.txt");
-            byte[] byteArray = Encoding.ASCII.GetBytes(bodyString);
-            MemoryStream stream = new MemoryStream(byteArray);
+            var byteArray = Encoding.ASCII.GetBytes(bodyString);
+            var stream = new MemoryStream(byteArray);
```
Comment (Aliandi) [inline tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioHelperTests.cs]: Just a question, why change to var?

Comment (GasparAcevedoZainSouthworks) [reply tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioHelperTests.cs]: Having `var` and a _explicit type_ (like _byte[]_) it's exactly the same in terms of performance, but it usage its encouraged for, for example, allowing to _adapt_ your code to changes in the methods you're calling. For example, if you left the _byte[]_ type but the _GetBytes_ method is refactored for returning another type, you must refactor your code; by the other hand, if you use `var`, this refactor wouldn't be necessary.

Comment (GasparAcevedoZainSouthworks) [reply tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioHelperTests.cs]: BTW, the type is inferred in `compile time` (not in runtime).

Comment (Aliandi) [reply tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioHelperTests.cs]: Super cool, I had no idea! Thanks!

**Comment by ceciliaavila** [comment]
Replaced with PR# 2462

**Pull request closed without merging**